lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -124,14 +127,30 @@ def add_shopper_interaction(post, payment, options={})
       def add_address(post, options)
         return unless post[:card] && post[:card].kind_of?(Hash)
         if (address = options[:billing_address] || options[:address]) && address[:country]
-          post[:card][:billingAddress] = {}
-          post[:card][:billingAddress][:street] = address[:address1] || 'N/A'
-          post[:card][:billingAddress][:houseNumberOrName] = address[:address2] || 'N/A'
-          post[:card][:billingAddress][:postalCode] = address[:zip] if address[:zip]
-          post[:card][:billingAddress][:city] = address[:city] || 'N/A'
-          post[:card][:billingAddress][:stateOrProvince] = address[:state] if address[:state]
-          post[:card][:billingAddress][:country] = address[:country] if address[:country]
+          post[:billingAddress] = {}
+          post[:billingAddress][:street] = address[:address1] || 'N/A'
+          post[:billingAddress][:houseNumberOrName] = address[:address2] || 'N/A'
+          post[:billingAddress][:postalCode] = address[:zip] if address[:zip]
+          post[:billingAddress][:city] = address[:city] || 'N/A'
+          post[:billingAddress][:stateOrProvince] = address[:state] if address[:state]
+          post[:billingAddress][:country] = address[:country] if address[:country]
         end
+        if shipping_address = options[:shipping_address] || address
+          post[:deliveryAddress] = {}
+          post[:deliveryAddress][:street] = shipping_address[:address1] if shipping_address[:address1]
+          post[:deliveryAddress][:houseNumberOrName] = shipping_address[:address2] if shipping_address[:address2]
+          post[:deliveryAddress][:postalCode] = shipping_address[:zip] if shipping_address[:zip]
+          post[:deliveryAddress][:city] = shipping_address[:city] if shipping_address[:city]
+          post[:deliveryAddress][:stateOrProvince] = shipping_address[:state] if shipping_address[:state]
+          post[:deliveryAddress][:country] = shipping_address[:country] if shipping_address[:country]
```

Comment:
_Nothing to do with code logic, just code cleanup:_

All of these `ifs` seem repetitive. What if he had something like:

```
delivery_address = {
  street: shipping_address[:address1],
  houseNumberOrName: shipping_address[:address2],
  postalCode: shipping_address[:zip],
  city: shipping_address[:city],
  stateOrProvince: shipping_address[:state],
  country: shipping_address[:country],
}
delivery_address.compact
```